[dez2003] iP

dez2003 opened the pull request:
### AlisaBot

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

What does AlisaBot _do_?

- keeps track of task progress
- organizes all of your tasks
- keeps ~~VERY~~ _SUPER_ IMPORTANT important details 

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

**ALL FOR FREE**

Features in AlisaBot:

- [ ] Managing tasks
- [ ] Reminders (coming soon)
- [ ] Managing event timings and deadlines (coming soon)

AlisaBot was built using Java. The main method looks like this:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

dez2003 committed:
Rename main class and project name to Alisa

dez2003 committed:
Add greeting and bye message

dez2003 committed:
Add echo feature

dez2003 committed:
Add Task class

dez2003 committed:
Build Task class with description feature

dez2003 committed:
Add support for list in main class

dez2003 committed:
Add methods in Task class to support mark as done feature

dez2003 committed:
Add support for mark as done commands in main class

dez2003 committed:
Ask Todo, Deadline, and Event class as types of Tasks

dez2003 committed:
Implement new task type Todo

dez2003 committed:
Implement new Task type Deadline

dez2003 committed:
Implement Task type Event

dez2003 committed:
Add auto testing for features

dez2003 committed:
Add AlisaException class to handle errors in chatbot

dez2003 committed:
Handle errors for current features

dez2003 committed:
Implement delete feature

dez2003 committed:
Create Storage class

dez2003 committed:
Add support for hard disk save feature

dez2003 committed:
Merge branch 'branch-Level-7'

dez2003 committed:
Modify Deadline and Event class to recognize dates and times for tasks

dez2003 committed:
Merge branch 'branch-Level-8'

dez2003 committed:
Implement Storage, Parser, TaskList, and Ui class to make code more OOP

dez2003 committed:
Implement Command class and its subclasses to write main logic

dez2003 committed:
Modify classes and implement main logic in Alisa class

dez2003 committed:
Merge branch 'branch-A-MoreOOP'

dez2003 committed:
Organize classes into packages

dez2003 committed:
Merge branch 'branch-A-Packages'

dez2003 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

dez2003 committed:
Add JUnit tests for Storage and Parser class

dez2003 committed:
Merge branch 'branch-A-JUnit'

dez2003 committed:
Modify build.gradle to create JAR file

dez2003 committed:
Merge branch 'branch-A-Jar'

dez2003 committed:
Add JavaDoc header comments for all classes

dez2003 committed:
Modify code to comply with coding standard

dez2003 committed:
Add feature that finds keyword in task description

dez2003 committed:
Merge branch 'branch-A-JavaDoc'

dez2003 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/alisa/task/TaskList.java
#	src/main/java/alisa/task/Todo.java

dez2003 committed:
Merge branch 'branch-Level-9'

kaixuan477 commented on gradle/wrapper/gradle-wrapper.properties:
You can use gitignore for the gradle files

kaixuan477 commented on src/main/java/alisa/command/Command.java:
```diff
@@ -0,0 +1,27 @@
+     * @param storage Storage that saves data into a file.
+     * @throws AlisaException If the operation fails.
+     */
+    public abstract void execute(TaskList taskList, Ui ui, Storage storage) throws AlisaException;
```
Impressive abstraction and use of Command class to handle user inputs!

kaixuan477 commented on src/test/java/alisa/ParserTest.java:
```diff
@@ -0,0 +1,60 @@
+
+public class ParserTest {
+    @Test
+    public void parse_markInput_markCommand() throws AlisaException {
```
Good naming of test methods!

kaixuan477 commented on src/main/java/alisa/Parser.java:
```diff
@@ -0,0 +1,45 @@
+package alisa;
+
+import alisa.command.*;
```
Following the coding standard it would be better to list imports explicitly instead of using wildcards.

dez2003 replied:
Oh nice I didn't notice that, thanks!

kaixuan477 reviewed (commented):
Nice job! Overall looks like a very comprehensive use of OOP and testing for Alisa, with just a minor change for coding standard.

marcusjhang commented on src/main/java/alisa/exception/AlisaException.java:
```diff
@@ -0,0 +1,13 @@
+package alisa.exception;
+
+public class AlisaException extends Exception {
```
class names - including exceptions all follow PascalCase, thats good

marcusjhang commented on src/main/java/alisa/command/UnmarkCommand.java:
```diff
@@ -0,0 +1,48 @@
+     * @return false.
+     */
+    @Override
+    public boolean isExit() {
```
method names all follow camelCase, very good

dez2003 committed:
Implement checkstyle to detect coding style violations

dez2003 committed:
Merge branch 'branch-A-CheckStyle'

dez2003 committed:
Add GUI to chatbot with JavaFX

dez2003 committed:
Merge branch 'branch-Level-10'

dez2003 committed:
Use assert feature in Alisa class

As long as there is input being received in the program, there should
always be a command to execute.

The assertion keyword in the run() method ensures that there is a
command to execute so that the program can run.

dez2003 committed:
Merge pull request #1 from dez2003/branch-A-Assertions

Use assert feature in Alisa class

dez2003 committed:
Rename toFileString() method to convertToFileString()

The method name wasn't used as a verb and didn't sufficiently describe
its function since there's another similar method called toString().

Let's rename the method by adding the word convert to further explain
what it does.

dez2003 committed:
Merge branch 'master' into branch-A-CodeQuality

dez2003 committed:
Merge pull request #2 from dez2003/branch-A-CodeQuality

Rename toFileString() method to convertToFileString()

dez2003 committed:
Add new edit feature to tasks

If a user wants to edit a specific part of a task, they would need to
delete the original task and then add a completely new task.

Let's add a new feature that can directly edit a part of an existing
task without having to use multiple commands. For instance, in an event
task, the edit command can allow just the end time to be updated.

Having an edit feature is preferable as it is easier to make
direct changes to tasks that still need to be on the task list.

dez2003 committed:
Merge pull request #4 from dez2003/branch-C-Update

Add new edit feature to tasks

dez2003 committed:
Modified the GUI to set up product website

dez2003 committed:
Modify GUI to set up product website

dez2003 committed:
Display welcome message in GUI

dez2003 committed:
Add user guide for Alisa chatbot

dez2003 committed:
Add and edit JavaDoc in Command classes

Since the EditCommand class was newly added, there are no JavaDoc
comments in any of its methods. The execute() method in Command class
was modified to return a String, so the JavaDoc comments for the other
classes are also not updated.

Not updating the JavaDoc comments properly would make it hard for other
developers seeing the code to understand its functionalities and how to
use the methods.

Let's put in updated JavaDoc comments for the Command class and its
subclasses.

dez2003 committed:
Add JavaDoc comments and edit editTask() method in TaskList class

There are no JavaDoc comments in the editTask() method. And in the
editTask(), the wrong task is being edited since the taskList index
starts with 0 instead of 1.

Not updating the JavaDoc comments properly would make it hard for other
developers to understand how to use editTask().

Let's put in updated JavaDoc comments for the TaskList class and correct
the index for the task to edit in editTask().

dez2003 committed:
Add JavaDoc comments and edit changeDueDate() in Deadline class

There are no JavaDoc comments in the changeDueDate() method. And in the
method, the DateTime format is incorrect since the month and hour
should be capitalized.

Not updating the JavaDoc comments properly would make it hard for other
developers to understand how to use changeDueDate(). Formatting the
date and time incorrectly will throw an exception since it can't
be parsed.

Let's put in updated JavaDoc comments for the Deadline class and correct
the DateTime format in changeDueDate().

dez2003 committed:
Add JavaDoc comments and edit constructor in Event class

There are no JavaDoc comments in the changeStartTime() and
changeEndTime() methods. And in the constructor, the DateTime format is
incorrect since the month and hour should be capitalized.

Not updating the JavaDoc comments properly would make it hard for other
developers to understand how to use changeStartTime() and
changeEndTime(). Formatting the date and time incorrectly will throw an
exception since it can't be parsed.

Let's put in updated JavaDoc comments for the Event class and correct
the DateTime format in the constructor.

dez2003 committed:
Add JavaDoc comments in Task class

There are no JavaDoc comments for the changeDescription() method.

Not updating the JavaDoc comments properly would make it hard for other
developers to understand how to use the changeDescription() method.

Let's add JavaDoc comments for changeDescription() in the Task class.
